Sources/Segment/Plugins/SegmentDestination.swift

```diff
@@ -120,20 +120,25 @@ public class SegmentDestination: DestinationPlugin {
             for url in data {
                 analytics.log(message: "Processing Batch:\n\(url.lastPathComponent)")
                 
-                let uploadTask = httpClient.startBatchUpload(writeKey: analytics.configuration.values.writeKey, batch: url) { (result) in
+                let uploadTask = httpClient.startBatchUpload(writeKey: analytics.configuration.values.writeKey, batch: url) { [weak self] (result) in
                     switch result {
                         case .success(_):
                             storage.remove(file: url)
-                            self.cleanupUploads()
-                        default:
-                            analytics.logFlush()
+                            self?.cleanupUploads()
+                        case let .failure(error):
+                            // Until https://github.com/segmentio/analytics-swift/issues/152 is resolved we should remove failing files since we run the risk of building up a huge list of pending files to upload.
+                            if error.connectivityError {
```

Comment:
hmhmhm, makes me a little nervous to discard all on network failure. (not sure when flush is triggered, think of short network failures / tunnel / tube / .. :-)) 

Thought about capping the `data` thing we iterate over - e.g. dropping everything > 50 but leaving the rest here untouched? (though we'd need to know the "order")

Reply:
It looks like flush is triggered quite frequently but here are a couple of key scenarios:
- Foregrounding the app
- Backgrounding the app
- At a somewhat set interval (I believe we have this set to every 10 seconds currently)

A slightly more sophisticated solution here would be tracking failures per file and then maybe deleting a file after X failures? 

I'd venture a guess only a handful of people are in the configuration where they are blocking api.segment.io so I erred on the side of the simplest solution that fixes the problem, while trying to articulate the trade offs that there would likely be some data log around our instrumentation due to this approach for the scenarios that you mention @diederich. It's a bit hard to quantify what the size of data loss would be for non-blocking members, but it's a tradeoff nonetheless. Would love @rgrunberg's opinion here as well since she will be directly impacted.

Reply:
Ideally
1) we'd attempt more than once before dropping a file (in case the connectivity problem is temporary)
and/or
2) we'd have some instrumentation of how often there are temporary network issues resulting in event deletion

That being said, I'm not too worried about partial data loss for members who are just experiencing temporary network issues, since our key metrics are fairly robust to that (e.g. if you open 200 tabs in a day and we drop 5% of those logs, you're still going to be counted as an active member). So if these other options seem difficult to implement, I don't think they should block getting a fix in for the main issue here.

Reply:
Thanks @rgrunberg!

@brianmichel in the spirit of simple / not changing - maybe, before we get into the `for url in data {`-loop:
```
if data.length > some-max {
   let filesToDrop = data.prefix(data.length - some-max)
   filesToDrop.forEach { storage.remove(file: $0.url) }
   data = data.suffix(some-max)
}
```
and then let the rest of it as is? So we'd just be dropping things in case the pile up, but otherwise keep the retry logic and flushing etc. as is. WDYT?

Reply:
I like it, let's give it a shot and see how it goes!

Reply:
Pushed up this solution instead of what I had.